runtime/monitor.go

```diff
@@ -0,0 +1,334 @@
+package runtime
+
+import (
+	"fmt"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/Sirupsen/logrus"
+	raven "github.com/getsentry/raven-go"
+	"github.com/pborman/uuid"
+	"github.com/taskcluster/statsum"
+	"github.com/taskcluster/taskcluster-worker/runtime/client"
+)
+
+// A Monitor is responsible for collecting logs, stats and error messages.
+type Monitor interface {
+	// Measure values in statsum
+	Measure(name string, value ...float64)
+	// Increment counters in statsum
+	Count(name string, value float64)
+	// Measure time of fn in statsum
+	Time(name string, fn func())
+
+	// Report error/warning to sentry and write to log, returns incidentId which
+	// can be included in task-logs, if relevant.
+	ReportError(err error, message ...interface{}) string
+	ReportWarning(err error, message ...interface{}) string
+
+	// Write log messages to system log
+	Debug(...interface{})
+	Debugln(...interface{})
+	Debugf(string, ...interface{})
+	Print(...interface{})
+	Println(...interface{})
+	Printf(string, ...interface{})
+	Info(...interface{})
+	Infoln(...interface{})
+	Infof(string, ...interface{})
+	Warn(...interface{})
+	Warnln(...interface{})
+	Warnf(string, ...interface{})
+	Error(...interface{})
+	Errorln(...interface{})
+	Errorf(string, ...interface{})
+	Panic(...interface{})
+	Panicln(...interface{})
+	Panicf(string, ...interface{})
+
+	// Create child monitor with given tags (tags don't apply to statsum)
+	WithTags(tags map[string]string) Monitor
+	WithTag(key, value string) Monitor
+	// Create child monitor with given prefix (prefix applies to everything)
+	WithPrefix(prefix string) Monitor
```

Comment:
So I think you have a fair point here...
We should have:
 1) `type Monitor interface {...}` as an interface
 2) `type monitorBase struct {monitorImpl}`  implements `Monitor`
 3) `type monitorImpl interface {...}` which specifies logic to be implemented by a monitor provider logic... 

If you really want I can refactor... In fact I'm tempted because it's nice :)
On the other hand it seems a bit overkill.. So long as we have `Monitor` as an interface we can always do things to clean up this stuff later.